pkg/cmd/server/api/types.go

```diff
@@ -610,10 +610,11 @@ type UserAgentDenyRule struct {
 
 // MasterNetworkConfig to be passed to the compiled in network plugin
 type MasterNetworkConfig struct {
-	NetworkPluginName  string
-	ClusterNetworkCIDR string
-	HostSubnetLength   uint32
-	ServiceNetworkCIDR string
+	NetworkPluginName    string
+	ClusterNetworkCIDR   string
+	ClusterNetworkConfig []ClusterNetworkEntry
```

Comment:
I'd probably go with just `ClusterNetworks`.

It would be nice if we could get rid of the old `ClusterNetworkCIDR` field, but I'm not sure how backward-compat works with startup config objects. Old master-config.yaml files need to keep working though, so whatever that requires...